File: hello_world.cs
//  System is a namespace which contains the commonly used types. It is specified with a 'using' System directive.
using System;

//  The 'namespace' keyword is used to define our own namespace. Here we are creating a namespace called HelloWorld.
//  namespace acts as a container which consists of classes, methods and other namespaces.
namespace HelloWorld
{
    //  The statement creates a class named - 'Hello' in C#.
    // Since, C# is an object-oriented programming language, creating a class is mandatory for the program's execution.
    public class Hello
    {
        //  Here 'static' keyword tells us that this method is accessible without instantiating the class.
        // The 'void' keyword tells that this method will not return anything.
        //  The execution of every C# program starts from the Main() method.
        public static void Main(string[] args)
        {
            // WriteLine() is a method of the 'Console' class defined in the System namespace.
            // This is the piece of code that prints Hello World! to the output screen.
            Console.WriteLine("Hello, World!");
        }
    }
}

// For more info check this microsoft's official website https://docs.microsoft.com/en-us/dotnet/csharp/
// For more info check this https://www.programiz.com/csharp-programming/hello-world
// For refernces to that check the site https://www.geeksforgeeks.org/hello-world-in-c-sharp/

// OUTPUT:
// Hello, World!

File: hello_world.go
// Official Website: https://golang.org/

/* package main - Every go file must start with the package name statement.
 * Packages are used to provide code compartmentalisation and reusability.
 * 'main' is the entry point for every go file.
 */
package main

// import "fmt" - The fmt package is imported and it will be used inside the main function to print text to the standard output.
import "fmt"

// func main() - The main is a special function. The program execution starts from the main function.
// The main function should always reside in the main package. The { and } indicate the start and end of the main function.
func main() {
	// fmt.Println("Hello World") - The Println function of the fmt package is used to write text to the standard output.
	fmt.Println("Hello, World!")
}

// ~$ go run hello_world.go
// Output: hello world

/* Learn more:
 * https://golangbot.com/learn-golang-series/
 */
